lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,356 @@
+        'jquery',
+        'underscore',
+        'gettext',
+        'moment',
+        'edx-ui-toolkit/js/utils/html-utils',
+        'js/learner_dashboard/models/course_entitlement_model',
+        'js/learner_dashboard/models/course_card_model',
+        'text!../../../templates/learner_dashboard/course_entitlement.underscore',
+        'text!../../../templates/learner_dashboard/verification_popover.underscore'
+    ],
+         function(
+             Backbone,
+             $,
+             _,
+             gettext,
+             moment,
+             HtmlUtils,
+             EntitlementModel,
+             CourseCardModel,
+             pageTpl,
+             verificationPopoverTpl
+         ) {
+             return Backbone.View.extend({
+                 tpl: HtmlUtils.template(pageTpl),
+                 verificationTpl: HtmlUtils.template(verificationPopoverTpl),
+
+                 events: {
+                     'change .session-select': 'updateEnrollBtn',
+                     'click .enroll-btn': 'handleEnrollChange',
+                     'click .popover-dismiss': 'hideVerificationDialog',
+                     'keydown .final-confirmation-btn': 'handleVerificationPopoverA11y'
+                 },
+
+                 initialize: function(options) {
+                     this.$el = $(this.el);
+
+                     // Set up models and reload view on change
+                     this.courseCardModel = new CourseCardModel();
+                     this.entitlementModel = new EntitlementModel({
+                         availableSessions: this.formatDates(JSON.parse(options.availableSessions)),
+                         entitlementUUID: options.entitlementUUID,
+                         currentSessionId: options.currentSessionId,
+                         userId: options.userId
+                     });
+                     this.listenTo(this.entitlementModel, 'change', this.render);
+
+                     // Grab URLs that handle changing of enrollment and entering a newly enrolled session.
+                     this.enrollUrl = options.enrollUrl;
+                     this.courseHomeUrl = options.courseHomeUrl;
+
+                     // Grab elements from the parent card that work with this view and bind associated events
+                     this.$triggerOpenBtn = $(options.triggerOpenBtn) || $(''); // Opens/closes session selection view
+                     this.$dateDisplayField = $(options.dateDisplayField) || $(''); // Displays current session dates
+                     this.$enterCourseBtn = $(options.enterCourseBtn) || $(''); // Button link to course home page
+                     this.$courseCardMessages = $(options.courseCardMessages) || $(''); // Additional session messages
+                     this.$courseTitleLink = $(options.courseTitleLink) || $(''); // Title link to course home page
+                     this.$courseImageLink = $(options.courseImageLink) || $(''); // Image link to course home page
```

Comment:
I know you wanted to avoid testing for the presence on an element, but I think it is dangerous to potentially create 6 jQuery objects that don't point to anything. This has potential to lead to issues down the line if/when this code is edited. Please update to remove defaulting to empty jQuery objects. Is there a use case for this View to work without any of these 6 elements? If not I don't think we need to be too defensive in checking for their presence.

Reply:
Yes, in the programs page there will not be a courseImageLink, for example, so we need to make sure that it is handled and treated as an empty jQuery object. I will remove them now though and they can be reinserted in the case that they aren't needed on the programs dashboard.

Reply:
Thanks. When updated in the future we shouldn't default to empty objects though. We should note in the code that these are optional elements and check for their existence before attempting to interact with them.